ZeeHomeScreen/Classes/Components/BasicContainerCell/UniversalCollectionViewCell.m

```diff
@@ -6,76 +6,45 @@
 //
 
 #import "UniversalCollectionViewCell.h"
-#import "CarouselViewController.h"
-#import <ZeeHomeScreen/ComponenttFactory.h>
 #import <ZeeHomeScreen/ZeeHomeScreen-Swift.h>
 
 @interface UniversalCollectionViewCell()
 
-@property (nonatomic, strong) NSLayoutConstraint *cellWidthConstraint;
-@property (nonatomic, strong) NSLayoutConstraint *cellHeightConstraint;
+@property (nonatomic) NSLayoutConstraint *cellWidthConstraint;
+@property (nonatomic) NSLayoutConstraint *cellHeightConstraint;
 
 - (void)updateFlexibilityConstraints:(ComponentModel *)componentModel;
 
 @end
 
 @implementation UniversalCollectionViewCell
 
-- (instancetype)initWithFrame:(CGRect)frame {
-    self = [super initWithFrame:frame];
-    if (self) {
-    
-    }
-    return self;
-}
-
 -(void)prepareForReuse {
     [super prepareForReuse];
-    
-    if ([self.componentViewController respondsToSelector:@selector(prepareComponentForReuse)]) {
-        [self.componentViewController prepareComponentForReuse];
+
+    if (self.parentHandlesReusingComponent) {
+        if ([self.componentViewController respondsToSelector:@selector(prepareComponentForReuse)]) {
+            [self.componentViewController prepareComponentForReuse];
+        }
+    } else {
+        [self.componentViewController removeViewFromParentViewController];
+        _componentViewController = nil;
     }
 }
 
 -(void)setComponentViewController:(UIViewController<ComponentProtocol> *)componentViewController {
-    _componentViewController = componentViewController;
-}
-
-- (UIViewController<ComponentProtocol> *)setComponentModel:(nullable ComponentModel *)componentModel
-                    model:(nullable NSObject *)model
-                     view:(nullable UIView *)view
-                 delegate:(nullable id<ComponentDelegate>)delegate
-     parentViewController:(nullable UIViewController *)parentViewController
-{
-    if (self.componentViewController == nil) {
-        self.componentViewController = [ComponenttFactory componentViewControllerWithComponentModel:componentModel
-            andModel:model
-            forView:view
-            delegate:delegate
-            parentViewController:parentViewController];
-    } else {
-        if ([self.componentViewController respondsToSelector:@selector(setComponentModel:)]) {
-            self.componentViewController.componentModel = componentModel;
-        }
-        if ([self.componentViewController respondsToSelector:@selector(rebuildComponent)]) {
-            [self.componentViewController rebuildComponent];
-        }
-    }
-    
-    if ([self.componentViewController respondsToSelector:@selector(setDelegate:)]) {
-        self.componentViewController.delegate = delegate;
+    if (_componentViewController != nil) {
```

Comment:
Release the previous view controller, when the new one comes in. It solves the problem with duplicating instances.